src/scenario/area.hpp

```diff
@@ -41,6 +42,17 @@ class cArea {
 		, terrain(dim, dim)
 		, maps(dim, boost::dynamic_bitset<>(dim))
 	{}
+
+	bool is_on_map(location loc) const
+	{
+		return loc.x < max_dim && loc.y < max_dim && loc.x >= 0 && loc.y >= 0;
```

Comment:
Uh, wait a second, isn't this... a little backwards? I mean, it should be <0 and >=max_dim, right?

Reply:
... no? wouldn't that be 'not_on_map'? Unless the dimension is negative and so are the coordinates

Reply:
I look at it now and it seems correct after all. I'm not sure what I was thinking when I wrote the original comment.